cmake/json_test.cmake

```diff
@@ -4,7 +4,21 @@ set(_json_test_cmake_list_file ${CMAKE_CURRENT_LIST_FILE})
 # download test data
 #############################################################################
 
-include(download_test_data)
+# if variable is set, use test data from given directory rather than downloading them
+if(JSON_TEST_DATA_DIRECTORY)
+    add_custom_target(download_test_data)
+    file(WRITE ${CMAKE_BINARY_DIR}/include/test_data.hpp "#define TEST_DATA_DIRECTORY \"${JSON_TEST_DATA_DIRECTORY}\"\n")
```

Comment:
For follow-up PR: Use `PROJECT_BINARY_DIR`.

Reply:
Since this is only relevant if someone wanted to build unit tests in an `add_subdirectory()` scenario, I don't think this matters.